lib/extractor/layer.ts

```diff
@@ -135,6 +201,48 @@ function getLayersContentAndArchiveManifest(
   };
 }
 
+function getOciLayersContentAndArchiveManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+  layers: Record<string, ExtractedLayers>,
+): { layers: ExtractedLayers[]; manifest: OciArchiveManifest } {
+  // filter empty layers
+  // get the layers content without the name
+  // reverse layers order from last to first
+
+  // get manifest file first
+  const manifest = getOciManifest(imageIndex, manifestCollection);
+  const filteredLayers = manifest.layers
+    .filter((layer) => Object.keys(layers[layer.digest]).length !== 0)
+    .map((layer) => layers[layer.digest])
+    .reverse();
+
+  return {
+    layers: filteredLayers,
+    manifest,
+  };
+}
+
+function getOciManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+): OciArchiveManifest {
+  if (imageIndex.manifests.length === 1) {
+    return manifestCollection[0];
```

Comment:
what happens if there is only one manifest and it is not linux amd64?

Reply:
😱That should have been removed when I've added the `error`.

Reply:
Ok, so I've updated it (and tested) so if we've got a platorm, return undefined if it doesn't match and if we don't have the platform, return the first one (platform is optional and only present when multiple manifests)

Reply:
should be verified with @garethr :)